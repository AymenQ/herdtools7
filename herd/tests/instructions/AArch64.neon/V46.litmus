AArch64 V46
(* Tests LD1/ST1 single structure, post-indexed immediate offset *)
{
uint128_t x;
0:X0=x; 0:X1=16; 0:X2=32;
}
 P0                    ;
 MOV W1,#16            ;
 STR W1,[X0]           ;
 MOV W2,#32            ;
 STR W2,[X0,#8]        ;
 LD1 {V0.S}[0],[X0],#4 ;
 ST1 {V0.S}[0],[X0],#4 ;
 LDR W3,[X0]           ;
forall(0:X3=32 /\ 0:X0!=x)
