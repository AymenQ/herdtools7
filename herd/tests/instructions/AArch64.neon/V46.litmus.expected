Test V46 Required
States 1
0:X0=x+8; 0:X3=32;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X3=32 /\ not (0:X0=x))
Observation V46 Always 1 0
Hash=b2e1a6ebd405791d7e4cfb1f11b5550a

